hal/pinmap.h

```diff
@@ -43,6 +43,12 @@ typedef struct {
     const int *peripheral;
 } PeripheralList;
 
+typedef struct {
+    int peripheral;    // Explicitly specified peripheral
```

Comment:
Wouldn't this better be a `void *`? Otherwise, it should at least be `intptr_t`.

Reply:
Maybe we should make it consistent with: `uint32_t pinmap_peripheral(PinName pin, const PinMap *map)` ?

Reply:
I haven't quite figured out how this is used in all cases, but I'm seeing it being cast to pointers, hence `intptr_t`. Anything that could avoid casts would be good...

Reply:
It depends on the target. Usually, it is cast to pointers to the target-specific control structures, so I would go for `void *`.